internal/provider/validators.go

```diff
@@ -441,3 +441,22 @@ func validateBoundaryPassword(v interface{}, path cty.Path) diag.Diagnostics {
 
 	return diagnostics
 }
+
+func validateVaultPluginType(v interface{}, path cty.Path) diag.Diagnostics {
+	var diagnostics diag.Diagnostics
+
+	err := vaultmodels.HashicorpCloudVault20201125PluginType(strings.ToUpper(v.(string))).Validate(strfmt.Default)
+	if err != nil {
+		enumList := regexp.MustCompile(`\[.*\]`).FindString(err.Error())
+		expectedEnumList := strings.ToLower(enumList)
+		msg := fmt.Sprintf("expected '%v' to be one of: %v", v, expectedEnumList)
```

Comment:
Sorry keep finding little things as I work on my own changes

Nit: this is a pattern that affects all our existing enum validations, but the way we currently validate also surfaces the "invalid" enum as an allowed value in our error message.

Encountered this when I was testing on my own branch, so I got an error message like
```
Error: expected '' to be one of: [http_proxy_option_invalid enabled disabled]

          with hcp_vault_cluster.test,
          on terraform_plugin_test.tf line 15, in resource "hcp_vault_cluster" "test":
          15: 	proxy_endpoint     = ""
```
when I passed the wrong proxy option.

Guessing you would get something similar with
```
expected %v to be one of: [plugin_type_invalid, secret, auth, database]
```

I don't think this is a blocking comment since it already affects so many of our other enums (all the MVU ones), but if you wanted to address this on yours, I ended up doing something like 

<details><summary>this example</summary>

```go
func validateVaultClusterProxyEndpoint(v interface{}, path cty.Path) diag.Diagnostics {
	var diagnostics diag.Diagnostics

	err := vaultmodels.HashicorpCloudVault20201125HTTPProxyOption(strings.ToUpper(v.(string))).Validate(strfmt.Default)
	if err != nil {
		enumList := regexp.MustCompile(`\[(.*)\]`).FindStringSubmatch(err.Error())
		expectedEnumList := strings.ToLower(enumList[1])

		// Remove invalid option from allowed list in error message
		expectedEnumList = strings.ReplaceAll(
			expectedEnumList,
			strings.ToLower(string(vaultmodels.HashicorpCloudVault20201125HTTPProxyOptionHTTPPROXYOPTIONINVALID)),
			"",
		)

		// Format as comma-separated list
		expectedEnumList = strings.Join(strings.Fields(expectedEnumList), ", ")

		msg := fmt.Sprintf("expected '%v' to be one of: %v", v, expectedEnumList)
		diagnostics = append(diagnostics, diag.Diagnostic{
			Severity:      diag.Error,
			Summary:       msg,
			Detail:        msg + " (value is case-insensitive).",
			AttributePath: path,
		})
	}

	return diagnostics
}

```

</details>

which gives an error message like `expected '' to be one of: enabled, disabled` instead of the one above.